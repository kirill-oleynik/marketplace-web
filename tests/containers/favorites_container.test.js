const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const FavoritesContainer = require('../../app/containers/favorites_container').default;

const fakeStore = (state = {}) => ({
  subscribe() {},
  dispatch() {},
  getState() { return state; }
});

const favorite = {
  id: 1,
  application_id: 1
};

const favorites = {
  ids: [1],
  byId: {
    1: favorite
  }
};

const application = {
  id: 1,
  title: 'Hello World'
};

const applications = {
  ids: [1],
  byId: {
    1: application
  }
};

describe('#render', () => {
  test('it renders correctly', () => {
    const store = fakeStore({
      favorites,
      applications
    });

    const component = shallow(
      <FavoritesContainer
        t={(translation) => translation}
      />,
      { context: { store } }
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#passedProps', () => {
  test('it passess props to component correctly', () => {
    const store = fakeStore({
      favorites,
      applications
    });

    const component = shallow(
      <FavoritesContainer
        t={(translation) => translation}
      />,
      { context: { store } }
    );

    const props = component.props();

    expect(props.favorites).toEqual([{ application, ...favorite }]);
    expect(props.removeFromFavorites).toBeInstanceOf(Function);
  });
});
