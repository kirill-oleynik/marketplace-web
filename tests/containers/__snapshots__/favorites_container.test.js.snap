// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<Translate(FavoritesContainer)
  dispatch={[Function]}
  favorites={
    Array [
      Object {
        "application": Object {
          "id": 1,
          "title": "Hello World",
        },
        "application_id": 1,
        "id": 1,
      },
    ]
  }
  t={[Function]}
/>
`;
